gslib/wildcard_iterator.py

```diff
@@ -583,16 +583,23 @@ class FileWildcardIterator(WildcardIterator):
   files in any subdirectory named 'abc').
   """
 
-  def __init__(self, wildcard_url, ignore_symlinks=False, logger=None):
+  def __init__(self,
+               wildcard_url,
+               exclude_pattern=None,
+               ignore_symlinks=False,
+               logger=None):
     """Instantiates an iterator over BucketListingRefs matching wildcard URL.
 
     Args:
       wildcard_url: FileUrl that contains the wildcard to iterate.
+      exclude_pattern: A regex of paths to ignore during iteration.
       ignore_symlinks: If True, ignore symlinks during iteration.
       logger: logging.Logger used for outputting debug messages during
               iteration. If None, the root logger will be used.
     """
     self.wildcard_url = wildcard_url
+    self.exclude_pattern = re.compile(
+        exclude_pattern) if exclude_pattern else None
```

Comment:
Recommended to perform explicit checks for None. i.e if exclude_pattern is not None else None

Reply:
Just realized the exclude_pattern is already a RegEx when it's passed in, so this whole line is being simplified to a simple assignment.